x-pack/plugins/snapshot_restore/public/application/components/restore_snapshot_form/steps/step_logistics/step_logistics.tsx

```diff
@@ -564,11 +568,34 @@ export const RestoreSnapshotStepLogistics: React.FunctionComponent<StepProps> =
           </EuiTitle>
         }
         description={
-          <FormattedMessage
-            id="xpack.snapshotRestore.restoreForm.stepLogistics.includeGlobalStateDescription"
-            defaultMessage="Restores templates that don’t currently exist in the cluster and overrides
-              templates with the same name. Also restores persistent settings."
-          />
+          <>
+            <FormattedMessage
+              id="xpack.snapshotRestore.restoreForm.stepLogistics.includeGlobalStateDescription"
+              defaultMessage="Restores templates that don’t currently exist in the cluster and overrides
+              templates with the same name. Also restores persistent settings and all system indices. {learnMoreLink}"
+              values={{
+                learnMoreLink: (
+                  <EuiLink target="_blank" href={docLinks.links.snapshotRestore.restoreSnapshotApi}>
+                    {i18n.translate(
+                      'xpack.snapshotRestore.restoreForm.stepLogistics.includeGlobalStateDocLink',
+                      { defaultMessage: 'Learn more.' }
+                    )}
+                  </EuiLink>
+                ),
+              }}
+            />
+
+            {/* Only display callout if include_global_state is enabled and the snapshot was created by ES 7.12+
+             * Note: Once we support features states in the UI, we will also need to add a check here for that
+             * See https://github.com/elastic/kibana/issues/95128 more details
+             */}
+            {includeGlobalState && semverGt(version, '7.12.0') && (
```

Comment:
Thanks for adding the explanation about features states, it's very helpful to understand the code